File: Acceloka.Api/Features/Entities/Ticket.cs
namespace Acceloka.Api.Features.Entities;

public class Ticket
{
    public string TicketCode { get; set; } = string.Empty;
    public string TicketName { get; set; } = string.Empty;
    public string CategoryName { get; set; } = string.Empty;
    public DateTime EventDate { get; set; }
    public Decimal Price { get; set; }
    public int Quota { get; set; }
}

File: acceloka-web/src/types/ticket.ts
export interface Ticket {
    ticketCode: string;
    ticketName: string;
    categoryName: string;
    eventDate: string; //ISO string format
    price: number;
    quota: number;
}

export interface TicketResponse {
    tickets: Ticket[];
    total: number;
}

export interface TicketFilter {
    search?: string;
    categoryName?: string;
    MinEventDate?: string;
    MaxEventDate?: string;
    MaxPrice?: number;
    OrderBy?: string;
    SortDirection?: "asc" | "desc";
    Page?: number;
    PageSize?: number;
}